### PR TITLE
Integration accuracy test (inspired by L2Projection integration method)

### PR DESCRIPTION
**Pull request by aprokop**
Continuation of #119 after unsuccessful force push.

### CODE REVIEW

**Comment by aprokop** [comment]
ok to test

**Comment by Rombur** [comment]
Jenkins didn't start :-/  Do a dummy push to the branch to trigger Jenkins again.

**Comment by aprokop** [comment]
I wonder if it only checks for @dalg24's "ok to test". Could you try it too?

**Comment by Rombur** [comment]
I can try but normally the test is triggered when the PR is created and by any push on a PR.

**Comment by Rombur** [comment]
ok to test

**Comment by dalg24** [comment]
ok to test

**Comment by aprokop** [comment]
Hmm, so apparently only @dalg24 can kick off the testing (at least for my PR)...

**Comment by Rombur** [comment]
well that sucks...

**Comment by aprokop** [comment]
I think this is a Jenkins configuration issue. Specifically, when #129 was submitted the Jenkins did not ask "can I test?" question. So it must be checking of who is submitting the patch.

**Comment by dalg24** [comment]
@aprokop What do you mean unsuccessful force push?  You are woking your own fork, you should be able to do that as often as you like...

**Comment by aprokop** [comment]
@dalg24 I updated the master and rebased the fork on top of it, that broke it. I think as long as the commit you use as your base is the same it's fine; as soon as it changes github closes it. 
I tried rebasing on top of the old hash but that did not work for me either.

**Comment by dalg24** [comment]
That's weird I do this all the time and I never had any issue.

**Comment by aprokop** [comment]
@sslattery @dalg24 Please review.

File: packages/Operators/test/reference_implementation/DTK_ReferenceHexMesh.cpp [hunk 1]
```diff
@@ -208,53 +248,19 @@ void ReferenceHexMesh::buildMesh(
...
-                          ( k + 1 ) * x_num_nodes * y_num_nodes;
-                DTK_CHECK( node_id < total_nodes );
-                entity_set->getEntity( node_id, 0, hex_nodes[6] );
-
-                // node 7
-                node_id = ( i ) + ( j + 1 ) * x_num_nodes +
-                          ( k + 1 ) * x_num_nodes * y_num_nodes;
-                DTK_CHECK( node_id < total_nodes );
-                entity_set->getEntity( node_id, 0, hex_nodes[7] );
+#define GET_NODE( local_id, i, j, k )                                          \
```
Comment (sslattery) [inline packages/Operators/test/reference_implementation/DTK_ReferenceHexMesh.cpp]: Why not create a function instead of using a macro?

Comment (sslattery) [inline packages/Operators/test/reference_implementation/DTK_ReferenceHexMesh.cpp]: or a lambda?

File: packages/Operators/test/tstIntegral.cpp [hunk 2]
```diff
@@ -0,0 +1,472 @@
...
+    {
+        DTK_REQUIRE( x_min < 2 && x_max > 2 );
+        double a = 3.1 / 2;
+        return 0.5 * ( ( x_min - a ) * ( x_min - a ) +
+                       ( x_max - a ) * ( x_max - a ) ) *
+               ( y_max - y_min ) * ( z_max - z_min );
+    }
+
+    // should not reach this
+    return 0.0;
```
Comment (sslattery) [inline packages/Operators/test/tstIntegral.cpp]: maybe throw then if we reach this point?

File: packages/Operators/test/reference_implementation/DTK_ReferenceHexMesh.cpp [hunk 3]
```diff
@@ -50,8 +52,6 @@
 #include <DTK_BasicEntitySet.hpp>
 #include <DTK_EntityCenteredField.hpp>
 
-#include <DTK_DBC.hpp>
```
Comment (aprokop) [inline packages/Operators/test/reference_implementation/DTK_ReferenceHexMesh.cpp]: This was moved to `.hpp` file. It probably should have been a part of the commit containing `id()` calls as to make them inline I put them into `hpp` while simultaneously doing some `DTK_REQUIRE` there.

File: packages/Operators/test/reference_implementation/DTK_ReferenceHexMesh.cpp (hunk 1 above)
Comment (aprokop) [inline packages/Operators/test/reference_implementation/DTK_ReferenceHexMesh.cpp]: Don't think a function is a good choice. But did not think about lambdas. Will try to implement those.

File: packages/Operators/test/tstIntegral.cpp (hunk 2 above)
Comment (aprokop) [inline packages/Operators/test/tstIntegral.cpp]: OK.

**Comment by aprokop** [comment]
Updated PR to address @sslattery comments.

**Comment by dalg24** [comment]
That's weird I had these "pending comments".

File: packages/Operators/test/tstL2Projection.cpp [hunk 4]
```diff
@@ -96,17 +96,20 @@ double integrateField( DataTransferKit::UnitTest::ReferenceHexMesh &mesh,
...
+
+    auto ghosted_field = mesh.ghostedNodalField( field.dimension() );
     Teuchos::RCP<DataTransferKit::FieldMultiVector> ghosted_vector =
         Teuchos::rcp(
             new DataTransferKit::FieldMultiVector( comm, ghosted_field ) );
     Tpetra::Import<DataTransferKit::FieldMultiVector::LO,
                    DataTransferKit::FieldMultiVector::GO>
         importer( vector->getMap(), ghosted_vector->getMap() );
     ghosted_vector->doImport( *vector, importer, Tpetra::REPLACE );
+    ghosted_vector->pushDataToApplication();
```
Comment (aprokop) [inline packages/Operators/test/tstL2Projection.cpp]: You need this to sync data in MultiVector and field.

**Pull request merged**